### PR TITLE
Adopt CircleCI instead of Travis

### PR DESCRIPTION
**Pull request by softr8**
* Move decorators to app/decorators
* Gemfile now supports master branches and the right rails version

Also took in consideration #2 bringing better support

### CODE REVIEW

**Review by Ilyeo — APPROVED** [review]
pretty good 👍

File: app/decorators/models/solidus_payment_method_by_zone/spree/order_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module SolidusPaymentMethodByZone
+  module Spree
+    module OrderDecorator
+      def available_payment_methods
+        @available_payment_methods ||= super.left_joins(:payment_method_zone)
+                                           .where(spree_payment_method_zones: {zone_id: [nil] + ::Spree::Zone.for_address(ship_address).pluck(&:id)})
```
Comment (dannharo) [inline app/decorators/models/solidus_payment_method_by_zone/spree/order_decorator.rb]: could you add spaces inside this hash:
`{ zone_id: [nil] + ::Spree::Zone.for_address(ship_address).pluck(&:id) })`

File: Gemfile [hunk 2]
```diff
@@ -2,32 +2,22 @@ source 'https://rubygems.org'
 
 git_source(:github) { |repo_name| "https://github.com/#{repo_name}.git" }
 
+
 branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
 gem 'solidus', github: 'solidusio/solidus', branch: branch
 
-if branch == 'master' || branch >= 'v2.3'
```
Comment (dannharo) [inline Gemfile:8]: Is it no longer necessary include the `rails` version?

Comment (softr8) [reply Gemfile:8]: Nope, it will install what solidus has as dependency (newest possible)

**Pull request merged**